workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncherWindow.cs

```diff
@@ -303,14 +303,68 @@ private void OnGUI()
 
                 if (manager.IsActive)
                 {
-                    EditorGUILayout.HelpBox(GetStatusMessage(), MessageType.Info);
+                    using (new EditorGUILayout.HorizontalScope())
+                    {
+                        EditorGUILayout.HelpBox(GetStatusMessage(), MessageType.Info);
+                        if (GUILayout.Button("Cancel", GUILayout.Height(38), GUILayout.Width(75)))
+                        {
+                            CancelCurrentTask();
+                        }
+                    }
+
                     var rect = EditorGUILayout.GetControlRect(false, 20);
                     DrawSpinner(Time.realtimeSinceStartup * 10, rect);
+
                     Repaint();
                 }
             }
         }
 
+        private void CancelCurrentTask()
+        {
+            var sb = new StringBuilder();
+
+            switch (manager.CurrentTask)
+            {
+                case UploadTask task:
+                    sb.Append("Are you sure you want to cancel uploading your assembly?");
+                    break;
+                case LaunchTask task:
+                    sb.Append("Are you sure you want to cancel launching your deployment?");
+                    break;
+                case ListTask task:
+                    sb.Append("Are you sure you want to cancel listing live deployments?");
+                    break;
+                case StopTask task:
+                    sb.Append("Are you sure you want to cancel stopping your deployment?");
+                    break;
+                case AuthTask task:
+                    Debug.Log("Cannot cancel an authenticated task.");
```

Comment:
```suggestion
                    Debug.Log("Cannot cancel authentication against SpatialOS.");
```

or is it possible to simply not show the cancel button for this?

Reply:
yes that works better - I'll do that 👍